test/src/e2e_vm_tests/test_programs/should_pass/language/is_reference_type/src/main.sw

```diff
@@ -20,7 +20,7 @@ fn main() -> bool {
     assert(!is_reference_type::<byte>());
     assert(!is_reference_type::<u64>());
 
-    assert(is_reference_type::<str[1]>());
+    assert(!is_reference_type::<str[1]>());
```

Comment:
If we go with this, then we should probably make it clear in the language that types that "fit in a word" are always copy types. That includes things like `str[1]`, `struct { u64 }`, `struct { struct { u64 } }`, `array[1]`, and so on. It is a bit weird, but I think this would make the specs a bit simpler where the optimization that we've been talking about is actually baked in the language and _has_ to happen.

Reply:
I think you're right.  We'd absolutely need regression tests to enforce it for all our types too.